src/stores/RegistrationStore.ts

```diff
@@ -0,0 +1,127 @@
+import { ref, computed } from 'vue'
+import type { IRegistration, ITimeRange } from '@/interfaces'
+import {
+  extractWithDescription,
+  checkForOverlap,
+  saveToStorage,
+  loadFromStorage,
+} from '@/helpers'
+
+import { defineStore } from 'pinia'
+
+export const useRegistrationStore = defineStore('registrationStore', () => {
+
+  const formattedRegistrations = ref<Map<string, IRegistration>>(new Map())
+  const errors = ref<string[]>([])
+
+  const registrationsArray = computed(() => Array.from(formattedRegistrations.value.values()))
+
+  function formatWithoutSaving(registrationsText: string | undefined) {
+    const tempDataString = loadFromStorage();
+
+    formatRegistrationsCallback(registrationsText);
+    const registrations = JSON.parse(JSON.stringify(registrationsArray.value));
```

Comment:
its, sadly, the JS way of copying value instead of a reference.
this `const registrations = registrationsArray.value` will mutate along with registrationsArray.value

JS is funky stuff

Reply:
Quick googling brought me to this: https://stackoverflow.com/questions/728360/how-do-i-correctly-clone-a-javascript-object which points at this https://developer.mozilla.org/en-US/docs/Web/API/structuredClone

Would that be applicable? I really dont like that workaround

Reply:
Alternatively there is lodash https://github.com/lodash/lodash

Reply:
Sadly structuredClone didnt work - Vue doesnt directly work with object, but rather proxy objects.
I did find this solution though: https://stackoverflow.com/a/77022014. Still feels like a hack

I don't want to install lodash, if this is the only place we're gonna use it, seems redundant. 

Anyway, I've pushe dthe code from the stackoverflow solution